pkg/arm/v10/resources.go

```diff
@@ -49,7 +48,7 @@ func (g *simpleGenerator) vnet() *network.VirtualNetwork {
 		Type:     to.StringPtr("Microsoft.Network/virtualNetworks"),
 		Location: to.StringPtr(g.cs.Location),
 	}
-	if features.PrivateLinkEnabled(g.cs) {
+	if g.cs.Properties.PrivateAPIServer {
```

Comment:
if ManagementSubnetCIDR != nil ?

Reply:
this is probably OK for now. I'm not quite straight in my mind about how we gate all these pieces, but it'll become clearer.